stixpy/coordinates/frames.py

```diff
@@ -0,0 +1,160 @@
+            coord.RepresentationMapping("lat", "Ty", u.arcsec),
+            coord.RepresentationMapping("distance", "distance"),
+        ],
+        coord.UnitSphericalRepresentation: [coord.RepresentationMapping('lon', 'Tx', u.arcsec),
+                                            coord.RepresentationMapping('lat', 'Ty', u.arcsec)]
+    }
+
+
+def stix_wcs_to_frame(wcs):
+    r"""
+    This function registers the coordinate frames to their FITS-WCS coordinate
+    type values in the `astropy.wcs.utils.wcs_to_celestial_frame` registry.
+
+    Parameters
+    ----------
+    wcs : astropy.wcs.WCS
+
+    Returns
+    -------
+    astropy.coordinates.BaseCoordinateFrame
+    """
+    if hasattr(wcs, "coordinate_frame"):
+        return wcs.coordinate_frame
+
+    # Not a STIX wcs bail out early
+    if set(wcs.wcs.ctype) != {'SXLN-TAN', 'SXLT-TAN'}:
+        return None
+
+    dateobs = wcs.wcs.dateobs
+
+    rsun = wcs.wcs.aux.rsun_ref
+    if rsun is not None:
+        rsun *= u.m
+
+    hgs_longitude = wcs.wcs.aux.hgln_obs
+    hgs_latitude = wcs.wcs.aux.hglt_obs
+    hgs_distance = wcs.wcs.aux.dsun_obs
+
+    observer = HeliographicStonyhurst(hgs_latitude * u.deg,
+                                      hgs_longitude * u.deg,
+                                      hgs_distance * u.m,
+                                      obstime=dateobs,
+                                      rsun=rsun)
+
+    frame_args = {'obstime': dateobs,
+                  'observer': observer,
+                  'rsun': rsun}
+
+    return STIXImaging(**frame_args)
+
+
+def stix_frame_to_wcs(frame, projection='TAN'):
+    r"""
+    For a given frame, this function returns the corresponding WCS object.
+    It registers the WCS coordinates types from their associated frame in the
+    `astropy.wcs.utils.celestial_frame_to_wcs` registry.
+
+    Parameters
+    ----------
+    frame : astropy.coordinates.BaseCoordinateFrame
+    projection : str, optional
+
+    Returns
+    -------
+    astropy.wcs.WCS
+    """
+    # Bail out early if not STIXImaging frame
+    if not isinstance(frame, STIXImaging):
+        return None
+
+    wcs = WCS(naxis=2)
+    wcs.wcs.aux.rsun_ref = frame.rsun.to_value(u.m)
+
+    # Sometimes obs_coord can be a SkyCoord, so convert down to a frame
+    obs_frame = frame.observer
+    if hasattr(obs_frame, 'frame'):
+        obs_frame = frame.observer.frame
+
+    wcs.wcs.aux.hgln_obs = obs_frame.lon.to_value(u.deg)
+    wcs.wcs.aux.hglt_obs = obs_frame.lat.to_value(u.deg)
+    wcs.wcs.aux.dsun_obs = obs_frame.radius.to_value(u.m)
+
+    wcs.wcs.dateobs = frame.obstime.utc.iso
+    wcs.wcs.cunit = ['arcsec', 'arcsec']
+    wcs.wcs.ctype = ['SXLN-TAN', 'SXLT-TAN']
+
+    return wcs
+
+
+# Remove once min version of sunpy has https://github.com/sunpy/sunpy/pull/7594
+astropy.wcs.utils.WCS_FRAME_MAPPINGS.insert(1, [stix_wcs_to_frame])
+astropy.wcs.utils.FRAME_WCS_MAPPINGS.insert(1, [stix_frame_to_wcs])
```

Comment:
Is this necessary anymore?  Is the above bug fixed in sunpy?

Reply:
yea but it not been released or backported yet